client/src/components/league-teams/league-teams.js

```diff
@@ -2,26 +2,31 @@ import ko from 'knockout';
 
 import template from 'components/league-teams/league-teams.html';
 
-import {leaguesList} from 'models/leagues-list';
-import {leagueTeams} from 'models/league-teams';
-import {favorites} from 'models/favorites';
+import {leaguesList} from 'constants/leagues-list';
+import {leagueTeams} from 'providers/league-teams';
+import {favorites} from 'providers/favorites';
 
-function TeamsViewModel() {
-    this.favorites = favorites;
+class TeamsViewModel {
+    constructor() {
+        this.favorites = favorites;
+        this.leagues = leaguesList;
 
-    this.toggleFavoriteState = name => {
+        this.selectedLeagueName = ko.observable(leaguesList()[0]);
+
+        this.selectedLeagueTeams = ko.observable(leagueTeams.get(this.selectedLeagueName()));
```

Comment:
Again you have an observable that has an observable array as its value, which seems to be redundant.